contracts/Vouching.sol

```diff
@@ -1,216 +1,599 @@
+    Challenge storage challenge_ = challenges_[_challengeID];
+    require(challenge_.answer == Answer.PENDING, "Given challenge was already answered");
+    require(msg.sender == owner(challenge_.entryID) || !_withinAnswerPeriod(challenge_), "Challenges can only be answered by the entry owner during the answer period");
+
+    challenge_.answer = Answer.REJECTED;
+    challenge_.answeredAt = now;
+    emit Rejected(_challengeID);
   }
 
-}
+  /**
+   * @dev Appeals a decision by the vouched entry owner to accept or reject a decision. Any ZEP token holder can
+   * perform an appeal, staking a certain amount of tokens on it.
+   */
+  function appeal(uint256 _challengeID) public existingChallenge(_challengeID) {
+    Challenge storage challenge_ = challenges_[_challengeID];
+    require(_withinAppealPeriod(challenge_), "The appeal period has ended");
+    require(challenge_.answer != Answer.PENDING, "Cannot appeal a not-answered challenge");
+    require(challenge_.appeal.appealer == address(0), "Given challenge was already appealed");
+    require(msg.sender != owner(challenge_.entryID), "The owner of a vouched entry can not appeal their own decision");
+
+    Appeal storage appeal_ = challenge_.appeal;
+    appeal_.appealer = msg.sender;
+    appeal_.createdAt = now;
+    appeal_.amount = challenge_.amount.mul(appealFee_).div(PCT_BASE);
+    emit Appealed(_challengeID, msg.sender, appeal_.amount);
+
+    token_.safeTransferFrom(msg.sender, this, appeal_.amount);
+  }
+
+  /**
+   * @dev Accepts an appeal on a challenge. Can only be called by the overseer.
+   */
+  function sustain(uint256 _challengeID) public onlyOverseer existingChallenge(_challengeID) {
+    Challenge storage challenge_ = challenges_[_challengeID];
+    Appeal storage appeal_ = challenge_.appeal;
+    require(challenge_.resolution == Resolution.PENDING, "Given challenge was already resolved");
+    require(appeal_.appealer != address(0), "Cannot sustain a not-appealed challenge");
+
+    challenge_.resolution = Resolution.SUSTAINED;
+    emit Sustained(_challengeID, overseer_);
+
+    uint256 i;
+    uint256 _blocked;
+    Voucher storage voucher_;
+    Entry storage entry_ = entries_[challenge_.entryID];
+
+    if (challenge_.answer == Answer.REJECTED) {
+      entry_.totalVouched = entry_.totalVouched.sub(challenge_.amount);
+
+      for(i = 0; i < entry_.vouchersAddresses.length; i++) {
+        voucher_ = entry_.vouchers[entry_.vouchersAddresses[i]];
+        _blocked = voucher_.blockedPerChallenge[_challengeID];
+        if (_blocked > uint256(0)) voucher_.vouched = voucher_.vouched.sub(_blocked);
+        // no need to reset the voucher's blocked amount for the challenge
+      }
+
+      token_.safeTransfer(appeal_.appealer, appeal_.amount.mul(2));
+      token_.safeTransfer(challenge_.challenger, challenge_.amount.mul(2).sub(appeal_.amount));
+    }
+    else {
+      uint256 profit = challenge_.amount.sub(appeal_.amount);
```

Comment:
If we don't add an extra profit to the appealer, this should be equal to the challenge amount

Reply:
Fixed after the [simplification](https://github.com/zeppelinos/zos-vouching/pull/1#issuecomment-458948159) we agreed on the appeals payouts process